### PR TITLE
PL-53: Add Development Environment Components section.

### PR DESCRIPTION
**Pull request by damontgomery**
## JIRA Ticket(s)

- See ticket [PL-53: Document Solr requirements](https://palantir.atlassian.net/browse/PL-53).

## Description

Add documentation to the README about development environment components.

## Testing instructions

View the README.

---

[Pull Request Guidelines](https://github.com/palantirnet/development_documentation/blob/master/guidelines/pr_review.md)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,6 +28,19 @@ The development environment is based on [palantirnet/the-vagrant](https://github
...
+The development environment inside the virtual machine will be automatically configured by Vagrant.
+
+You may be interested in creating your own development environment and comparing it against the tested version.
+
+* Operating System: Ubuntu 64bit version 16.04.4 LTS
+* Web Host: Apache HTTP version 2.x
+* Database: MySQL version 5.7.21
+* PHP version 7.1.16
+* Search Server: Apache Solr version 4.5.1
+* Web Application: Drupal versions 7.61 and 8.6.2
```
Comment (agentrickard) [inline README.md]: Are we still running 8.6.2? We should upgrade to 8.6.5.

Comment (damontgomery) [reply README.md]: Can you open a story for that?

File: README.md [hunk 2]
```diff
@@ -5,6 +5,7 @@ This is the development repository for Federated Search Demo environment. It con
 ## Table of Contents
 
 * [Development Environment](#development-environment)
+* [Development Environment Componentes](#development-environment-components)
```
[the remark this answers is not in the document]
Comment (damontgomery) [reply README.md]: Spelling. :( Nice catch.

File: README.md (hunk 1 above)
Comment (agentrickard) [reply README.md]: https://palantir.atlassian.net/browse/PL-54

**Pull request merged**